go-client/client/client.go

```diff
@@ -0,0 +1,312 @@
+// Tables that are created by other clients or in the web UI are not listed here automatically.
+// Tables that are created in scripts run by this client, however, are immediately available,
+// and will be added to/removed from the list as soon as the script finishes.
+// FetchTables can be used to update the list to reflect what tables are currently available
+// from other clients or the web UI.
+// Calling FetchTables with a FetchRepeating argument will continually update this list in the background.
+func (client *Client) ListOpenableTables() []string {
+	client.tablesLock.Lock()
+	defer client.tablesLock.Unlock()
+
+	var result []string
+	for id := range client.tables {
+		if id.appId == "scope" {
+			result = append(result, id.fieldName)
+		}
+	}
+	return result
+}
+
+// newTicketNum returns a new ticket number that this client has not used before.
+// This function is thread-safe.
+func (client *Client) newTicketNum() int32 {
+	nextTicket := atomic.AddInt32(&client.nextTicket, 1)
+	if nextTicket <= 0 {
+		// If you ever see this panic... what are you doing?
+		panic("out of tickets")
+	}
+
+	return nextTicket
+}
+
+// newTicket returns a new ticket that this client has not used before.
+// This function is thread-safe.
+func (client *Client) newTicket() ticketpb2.Ticket {
+	id := client.newTicketNum()
+
+	return client.makeTicket(id)
+}
+
+// makeTicket turns a ticket ID into a ticket.
+// This function is thread-safe.
+func (client *Client) makeTicket(id int32) ticketpb2.Ticket {
+	bytes := []byte{'e', byte(id), byte(id >> 8), byte(id >> 16), byte(id >> 24)}
+
+	return ticketpb2.Ticket{Ticket: bytes}
+}
+
+// ExecQuery executes a query on the server and returns the resulting tables.
+//
+// If this function completes successfully,
+// the number of tables returned will always match the number of query nodes passed.
+// The first table in the returned list corresponds to the first node argument,
+// the second table in the returned list corresponds to the second node argument,
+// etc.
+//
+// This may return a QueryError if the query is invalid.
+func (client *Client) ExecQuery(ctx context.Context, nodes ...QueryNode) ([]*TableHandle, error) {
+	return execQuery(client, ctx, nodes)
+}
+
+// Close closes the connection to the server and frees any associated resources.
+// Once this method is called, the client and any TableHandles from it cannot be used.
+func (client *Client) Close() error {
+	client.lock.Lock()
+	defer client.lock.Unlock()
+
+	client.sessionStub.Close()
+	client.appStub.Close()
+	if client.grpcChannel != nil {
+		client.grpcChannel.Close()
+		client.grpcChannel = nil
+	}
+	// This is logged because most of the time this method is used with defer,
+	// which will discard the error value.
+	err := client.flightStub.Close()
+	if err != nil {
+		log.Println("unable to close client:", err.Error())
+	}
+	return err
+}
+
+// withToken attaches the current session token to a context as metadata.
+// This is thread-safe.
```

Comment:
As stated in the package description, all methods should be thread-safe unless stated otherwise, so I removed all of these redundant/misleading comments.
I believe all methods that are not thread-safe either make a note of what lock should be held, or explicitly state that they are not thread-safe (and in what way).